sycl/include/sycl/ext/intel/experimental/pipes.hpp

```diff
@@ -38,138 +39,222 @@ class pipe<_name, _dataT, _min_capacity, _propertiesT,
   // Non-blocking pipes
   // Reading from pipe is lowered to SPIR-V instruction OpReadPipe via SPIR-V
   // friendly LLVM IR.
-  template <class... _Params> static _dataT read(bool &_Success) {
+  template <typename _functionPropertiesT =
+                decltype(sycl::ext::oneapi::experimental::properties{})>
+  static _dataT read(bool &_Success) {
 #ifdef __SYCL_DEVICE_ONLY__
-    static constexpr auto _anchor_id =
-        _GetValue<int32_t, latency_anchor_id, _Params...>::value;
-    static constexpr auto _constraint =
-        _GetValue3<int32_t, type, int32_t, latency_constraint,
-                   _Params...>::value;
-
-    static constexpr int32_t _target_anchor = std::get<0>(_constraint);
-    static constexpr type _control_type = std::get<1>(_constraint);
-    static constexpr int32_t _cycle = std::get<2>(_constraint);
-    int32_t _type = 0; // Default: _control_type == type::none
-    if constexpr (_control_type == type::exact) {
-      _type = 1;
-    } else if constexpr (_control_type == type::max) {
-      _type = 2;
-    } else if constexpr (_control_type == type::min) {
-      _type = 3;
+    using _latency_anchor_id_prop = typename GetOrDefaultValT<
+        _functionPropertiesT,
+        sycl::ext::oneapi::experimental::latency_anchor_id_key,
```

Comment:
Why is it legal without `using namespace sycl::ext`?

Reply:
Because they share that namespace. This is in `sycl::ext::intel::experimental::` while the properties are in `sycl::ext::oneapi::experimental::`, so they are both in the `sycl::ext` namespace.

Reply:
Ah, I see. I'll update the code then!